lib/wallet_sweeper.js

```diff
@@ -428,12 +428,33 @@ WalletSweeper.prototype.discoverWalletFunds = function(increment, cb) {
 
                                             //get the unspent outputs for this batch of addresses
                                             return self.utxoFinder.getUTXOs(_.keys(batch)).then(function(utxos) {
+                                                // Do not evaluate 0-confirmation UTXOs
+                                                // This would include double spends and other things Insight happily accepts
+                                                // (and keeps in mempool - even when the parent UTXO gets spent otherwise)
+                                                for (var address in utxos) {
+                                                    if (utxos.hasOwnProperty(address) && Array.isArray(utxos[address])) {
+                                                        var utxosPerAddress = utxos[address];
+                                                        // Iterate over utxos per address
+                                                        for (var idx = 0; idx < utxosPerAddress.length; idx++) {
+                                                            if (utxosPerAddress[idx] &&
+                                                                'confirmations' in utxosPerAddress[idx]
+                                                                && utxosPerAddress[idx]['confirmations'] === 0) {
```

Comment:
It seems excessive to not include 0conf utxos all the time, maybe this could go in the recovery tool? seems like something we might tweak depending on results and better to not have to release the SDK as well for that